BrowserKit/Sources/Redux/Store.swift

```diff
@@ -68,15 +70,29 @@ public class Store<State: StateType>: DefaultDispatchStore {
     }
 
     public func dispatch(_ action: Action) {
-        guard Thread.isMainThread && !actionRunning else {
+        logger.log("Dispatched action: \(action.displayString())", level: .info, category: .redux)
+
+        guard Thread.isMainThread else {
             DispatchQueue.main.async { [weak self] in self?.dispatch(action) }
             return
         }
 
-        logger.log("Dispatched action: \(action.displayString())", level: .info, category: .redux)
+        actionQueue.append(action)
+        processQueuedActions()
+    }
 
-        actionRunning = true; defer { actionRunning = false }
+    private func processQueuedActions() {
+        guard !isProcessingActions else { return }
+        isProcessingActions = true
+        while !actionQueue.isEmpty {
+            let action = actionQueue.first!
+            actionQueue.remove(at: 0)
+            executeAction(action)
```

Comment:
```suggestion
            let action = actionQueue.removeFirst()
            executeAction(action)
```
nit: cleaner, no force unwrapping, and does the exact same thing. The LISPer in me would say just do `executeAction(actionQueue.removeFirst())` but I'm good either way.